db/active_replicator_checkpointer.go

```diff
@@ -258,9 +266,10 @@ func (c *Checkpointer) _updateCheckpointLists() (safeSeq *SequenceID) {
 	base.TracefCtx(c.ctx, base.KeyReplicate, "checkpointer: _updateCheckpointLists(expectedSeqs: %v, processedSeqs: %v)", c.expectedSeqs, c.processedSeqs)
 	base.TracefCtx(c.ctx, base.KeyReplicate, "Inside update checkpoint lists")
 
-	c.stats.ExpectedSequenceLen = len(c.expectedSeqs)
-	c.stats.ProcessedSequenceLen = len(c.processedSeqs)
-
+	if c.dbStats != nil {
+		c.dbStats.ProcessedSequenceLen.Set(int64(len(c.processedSeqs)))
```

Comment:
Is the alternative approach of storing pointers in CheckpointerStats and not having to keep a c.dbStats reference around better? I think it simplifies the code setting the stats values as all of them will be in the same place (`c.stats`)

Reply:
Now we've done that change we don't need `cb.dbStats` to access... we can continue to use `c.stats.ProcessedSequenceLen.Set(...)` directly